update section 5 (#419)
rancher/rke2: docs/cis_self_assessment.md

Inappropriate access to secrets stored within the Kubernetes cluster can allow for an attacker to gain additional access to the Kubernetes cluster or external resources whose credentials are stored as secrets.
</details>

> **Result:** Not Scored

Reviewer: ```suggestion
**Result:** Not Scored - Operator Dependent
```

> **Remediation:**

Reviewer: A few things:
1. The title for 5.1.2 is wrong. Looks like i copy/pasted in the title from 5.1.1. It shoiuld be:
```
Minimize access to secrets (Not Scored)
```
2. I cannot comment on it bc its too far up, but **5.1.1** needs a remediation. Just put something simple, like "RKE2 does not make inappropriate use of the cluster-admin role. Operators must audit their workloads of additional usage. See the CIS Benchmark guide for more details."
3. This (**5.1.2**) needas remediation too. How about "RKE2 limits its use of secrets for the system components appropriately, but operators must audit the use of secrets by their workloads. See the CIS Benchmark guide for more details."

The principle of least privilege recommends that users are provided only the access required for their role and nothing more. The use of wildcard rights grants is likely to provide excessive rights to the Kubernetes API.
</details>

**Result:** Not Scored - Operator Dependent

**Audit:**
Run the below command on the master node.

```bash
# Retrieve the roles defined across each namespaces in the cluster and review for wildcards
/var/lib/rancher/rke2/bin/kubectl get roles --all-namespaces -o yaml

# Retrieve the cluster roles defined in the	cluster	and	review for wildcards
/var/lib/rancher/rke2/bin/kubectl get clusterroles -o yaml
```

Verify that there are not wildcards in use.

> **NOTE** roles check passes however clusterroles does not

Reviewer: can delete. there are just built-in roles that have wildcards, which is fine

> **Remediation:**

Reviewer: ```suggestion
**Remediation:** Operators should review their workloads for proper role usage. See the CIS Benchmark guide for more details.
```

The ability to create pods in a cluster opens up possibilities for privilege escalation and should be restricted, where possible.
</details>

> **Result:** Not Scored

Reviewer: ```suggestion
**Result:** Not Scored - Operator Dependent
```

> **Remediation:**

Reviewer: ```suggestion
**Remediation:** Operators should review who has access to create pods in their cluster. See the CIS Benchmark guide for more details.
```

Avoiding mounting these tokens removes this attack avenue.
</details>

**Result:** Not Scored - Operator Dependent

**Remediation:**

> The pods launched by RKE2 are part of control plane and generally need access to communicate with the API server, thus this control does not apply to them. Operators should review their workloads and take steps to modify the definition of pods and service accounts which do not need to mount service account tokens to disable it.

Reviewer: ```suggestion
The pods launched by RKE2 are part of the control plane and generally need access to communicate with the API server, thus this control does not apply to them. Operators should review their workloads and take steps to modify the definition of pods and service accounts which do not need to mount service account tokens to disable it.
```

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
/var/lib/rancher/rke2/bin/kubectl get psp -o json | jq .items[] | jq -r 'select((.spec.hostPID == null) or (.spec.hostPID == false))' | jq .metadata.name | wc -l | xargs -I {} echo '--count={}'
```

Verify that the returned count is 1.

**Remediation:**

> RKE2 sets the `spec.HostPID` value to false explicitly in the (PodSecurityPolicySpec)[https://github.com/kubernetes/kubernetes/blob/cc68a78ef52a37f66d6e0c1569998e271470dc3a/staging/src/k8s.io/api/policy/v1beta1/types.go#L186]. The value of `false` is the default value for this field and due to how Go uses zero value and the field's JSON omit empty tag, this will not show up in the results. No manual remediation is needed.

Reviewer: ```suggestion
RKE2 sets the `spec.HostPID` value to false explicitly in. When reviewing PSPs, note that the Kubernetes API only displays this field if it is explicitly set to true. No manual remediation is needed.
```

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
/var/lib/rancher/rke2/bin/kubectl get psp -o json | jq .items[] | jq -r 'select((.spec.allowPrivilegeEscalation == null) or (.spec.allowPrivilegeEscalation == false))' | jq .metadata.name | wc -l | xargs -I {} echo '--count={}'
```

Verify that the returned count is 1.

**Remediation:**

> RKE2 sets the `allowPrivilegeEscalation` value to false explicitly for the PSP it creates. When reviewing PSPs, note that the Kubernetes API only displays this field if it is explicitly set to true. No manual remediation is needed.

Reviewer: This one shows up for me in the spec. does it for you?
```suggestion
RKE2 sets the `allowPrivilegeEscalation` value to false explicitly for the PSP it creates. No manual remediation is needed.
```

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
/var/lib/rancher/rke2/bin/kubectl get psp -o json | jq .items[] | jq -r 'select((.spec.allowPrivilegeEscalation == null) or (.spec.allowPrivilegeEscalation == false))' | jq .metadata.name | wc -l | xargs -I {} echo '--count={}'
```

Verify that the returned count is 1.

**Remediation:**

> RKE2 sets the `spec.runAsUser.Rule` value to `MustRunAsNonRoot` in the (PodSecurityPolicySpec)[https://github.com/kubernetes/kubernetes/blob/cc68a78ef52a37f66d6e0c1569998e271470dc3a/staging/src/k8s.io/api/policy/v1beta1/types.go#L193]. No manual remediation is needed.

Reviewer: I dont want to link to k8s code. Shouldnt expect operators to read go code.
```suggestion
RKE2 sets the `runAsUser.Rule` value to `MustRunAsNonRoot` in the PodSecurityPolicy that it creates. No manual remediation is needed.
```

If you need to run containers with this capability, this should be defined in a separate PSP and you should carefully check RBAC controls to ensure that only limited service accounts and users are given permission to access that PSP.
</details>

> **Result:** Not Scored - Operator Dependent

Reviewer: Can we say we pass for this one?

**Audit**
Run the below command on the master node.

```bash
/var/lib/rancher/rke2/bin/kubectl get psp global-restricted-psp -o json | jq .spec.requiredDropCapabilities[]

> ```

Reviewer: ```suggestion
```
Verify the value is `"ALL"`. 
```

> **Remediation:**

Reviewer: ```suggestion
Verify the value is `"ALL"`.

**Remediation:**
```

Reviewer: that suggestion came out funky. hopefully you get what im trying to do.
